cl/recap/api_serializers.py

```diff
@@ -152,6 +153,75 @@ def validate(self, attrs):
         return attrs
 
 
+class EmailProcessingQueueSerializer(serializers.ModelSerializer):
+    uploader = serializers.HiddenField(
+        default=serializers.CurrentUserDefault(),
+    )
+    court = serializers.PrimaryKeyRelatedField(
+        queryset=Court.federal_courts.all(),
+        html_cutoff=500,  # Show all values in HTML view.
+        required=True,
+    )
+    mail = serializers.JSONField(write_only=True)
+    receipt = serializers.JSONField(write_only=True)
+    recap_document = serializers.HyperlinkedRelatedField(
+        many=False,
+        read_only=True,
+        view_name="recapdocument-detail",
+        style={"base_template": "input.html"},
+    )
+
+    class Meta:
+        model = EmailProcessingQueue
+        fields = "__all__"
+        read_only_fields = (
+            "error_message",
+            "status",
+            "recap_documents",
+        )
+
+    def validate(self, attrs):
+        court_id = attrs["court"].pk
+        mail = attrs["mail"]
+        receipt = attrs["receipt"]
+
+        all_court_ids = Court.federal_courts.all_pacer_courts().values_list(
+            "pk", flat=True
+        )
+
+        if court_id not in all_court_ids:
+            raise ValidationError(
+                "%s is not a PACER court ID." % attrs["court"].pk
+            )
+
+        for attr_name in ["timestamp", "source", "destination", "headers"]:
+            if (
+                mail.get(attr_name) is None
+                or mail.get(attr_name) == "undefined"
+            ):
+                raise ValidationError(
+                    "The JSON value at key 'mail' should include '%s'."
+                    % attr_name
+                )
+
+        for attr_name in ["timestamp", "recipients"]:
+            if (
+                receipt.get(attr_name) is None
+                or receipt.get(attr_name) == "undefined"
+            ):
+                raise ValidationError(
+                    "The JSON value at key 'receipt' should include '%s'."
+                    % attr_name
+                )
+
+        return attrs
+
+    def create(self, validated_data):
+        validated_data.pop("mail", None)
+        validated_data.pop("receipt", None)
+        return super().create(validated_data)
```

Comment:
This throws away the mail and receipt data for now, right? Are youplanning it for this PR or for later?

Reply:
They get validated, but then we pull the actual JSON from the request in the view. Since they are not properties on the model, the serializer does not like it if they are in the final version. This was the advice I found to deal with non-model properties, but open to something else if you prefer. The serializer pattern is a bit more rigid than I would typically do, but I am trying to adhere to convention.

Reply:
Ah, OK. Maybe just a comment to that effect then, so we can remember in the future that this is a weird DRF thing?